.eslintrc.json

```diff
@@ -40,6 +40,8 @@
         "@typescript-eslint/no-non-null-assertion": "error",
         "chai-friendly/no-unused-expressions": "error",
         "prettier/prettier": "error",
+        "no-shadow": "off",
+        "@typescript-eslint/no-shadow": "warn",
```

Comment:
The vast majority of these lint changes address cases where AirBnB's defaults behave incorrectly in a TypeScript context (e.g. mistaking type identifiers for runtime identifiers, or not finding type identifiers). In some cases, there's an appropriate `@typescript-eslint` equivalent, in others the type checker itself addresses the equivalent. There are also a few rules changed in the demo to address minor differences between React JSX and Solid JSX.